portfolio/src/main/webapp/script.js

```diff
@@ -29,3 +32,18 @@ function topFunction() {
   document.documentElement.scrollTop = 0;
 }
 
+var getData;
+// External Citation:
+// https://www.quora.com/How-can-I-make-a-comment-box-in-HTML
+(getData = function getData(){
```

Comment:
You shouldn't need the `getData` function which you call immediately after declaring. I believe this will just work as:

```
fetch('/data').then(response => response.json()).then((data) => {
        console.log(data[0]);
        var displayComments = document.getElementById('display-comments');
        var comment;
        for(comment in data){
            console.log(data[comment]);
            displayComments.innerHTML += "" + data[comment] + "<br/>";
        }
    });
```

Since the function is simply running that code. Let me know if you would like me to explain further.

Reply:
It can also be helpful to the reader to say in a comment what `data` will look like.

i.e.
```
// data is expected to be an array of strings.
```

Also make sure not to do `data[0]` without checking the size, otherwise it'd be an array out of bounds error. It's quite helpful to always check array sizes and any data that may error out immediately in the code, i.e.:

```
fetch('/data').then(response => response.json()).then((data) => {
      if (...check for possible errors or empty response issues...) {
            ...handle errors...
            return;
      }
      ...rest of your code...
});
```

Reply:
done